okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -580,9 +580,9 @@ public void run() {
         while (frameReader.nextFrame(this)) {
         }
       } catch (IOException ioe) {
-        // We call onError instead of onIoException here, because OkHttp wraps many protocol errors
-        // as IOException, we should send GoAway for such errors.
-        onError(ErrorCode.PROTOCOL_ERROR, ioe.getMessage());
+        // We send GoAway here because OkHttp wraps many protocol errors as IOException.
+        frameWriter.goAway(0, ErrorCode.PROTOCOL_ERROR, new byte[0]);
```

Comment:
If there is a message in the `ioe`, we could send it in the GOAWAY. Probably a good idea. No need to do it now, but add a TODO.